### PR TITLE
configs: actually enable and disable on setup

### LINKED ISSUE #272: Modules are not enabled on startup
**Describe the bug**
I used the configuration example in the readme which should enable all the modules by default, but they are not. I have to run TSBufEnable on all the modules to enable them

**To Reproduce**
1. Configure all the modules to be enabled by default
2. Try using a module feature (textobjects)
3. The module doesn't work
4. Run TSBufEnable textobjects
5. The module now works

**Expected behavior**
The module should be enabled without running TSBufEnable

**Output of `:checkhealth nvim_treesitter`**

<details>
<code>

health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `git` executable found.
  - OK: `cc` executable found.

## elm parser healthcheck
  - OK: elm parser found.
  - WARNING: No `highlights.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c parser healthcheck
  - OK: c parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## java parser healthcheck
  - OK: java parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for java
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## python parser healthcheck
  - OK: python parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## dart parser healthcheck
  - OK: dart parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## lua parser healthcheck
  - OK: lua parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## ocaml parser healthcheck
  - OK: ocaml parser found.
  - WARNING: No `highlights.scm` query found for ocaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for ocaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for ocaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## go parser healthcheck
  - OK: go parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## nix parser healthcheck
  - OK: nix parser found.
  - WARNING: No `highlights.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## yaml parser healthcheck
  - OK: yaml parser found.
  - WARNING: No `highlights.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## json parser healthcheck
  - OK: json parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## jsdoc parser healthcheck
  - OK: jsdoc parser found.
  - WARNING: No `highlights.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## markdown parser healthcheck
  - OK: markdown parser found.
  - WARNING: No `highlights.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## julia parser healthcheck
  - OK: julia parser found.
  - WARNING: No `highlights.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## html parser healthcheck
  - OK: html parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for html
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## typescript parser healthcheck
  - OK: typescript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## swift parser healthcheck
  - OK: swift parser found.
  - WARNING: No `highlights.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## query parser healthcheck
  - OK: query parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for query
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## cpp parser healthcheck
  - OK: cpp parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## ruby parser healthcheck
  - OK: ruby parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ruby
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## vue parser healthcheck
  - OK: vue parser found.
  - WARNING: No `highlights.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## scala parser healthcheck
  - OK: scala parser found.
  - WARNING: No `highlights.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## tsx parser healthcheck
  - OK: tsx parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## rust parser healthcheck
  - OK: rust parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## bash parser healthcheck
  - OK: bash parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for bash
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c_sharp parser healthcheck
  - OK: c_sharp parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for c_sharp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `textobjects.scm` found.

## rst parser healthcheck
  - OK: rst parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## css parser healthcheck
  - OK: css parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## php parser healthcheck
  - OK: php parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for php
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## javascript parser healthcheck
  - OK: javascript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.

## haskell parser healthcheck
  - OK: haskell parser found.
  - WARNING: No `highlights.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## toml parser healthcheck
  - OK: toml parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for toml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## regex parser healthcheck
  - OK: regex parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

</code>
</details>

**Output of `nvim --version`**
```
NVIM v0.5.0-nightly
Build type: Release
LuaJIT 2.0.5
Compilation: /Users/laxmanso/.local/bin/homebrew/Library/Homebrew/shims/mac/super/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/tmp/neovim-20200807-86806-5koi6w/build/config -I/tmp/neovim-20200807-86806-5koi6w/src -I/Users/laxmanso/.local/bin/homebrew/include -I/tmp/neovim-20200807-86806-5koi6w/deps-build/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include -I/Users/laxmanso/.local/bin/homebrew/opt/gettext/include -I/tmp/neovim-20200807-86806-5koi6w/build/src/nvim/auto -I/tmp/neovim-20200807-86806-5koi6w/build/include
Compiled by laxmanso@A19723LaxmanSo

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/Users/laxmanso/.local/bin/homebrew/Cellar/neovim/HEAD-f26df8b/share/nvim"

Run :checkhealth for more info

```
This is my treesitter config

```
require'nvim-treesitter.configs'.setup {
	highlight = {
		enable = true
	},
	refactor = {
		highlight_definitions = {
			enable = true
		},
		highlight_current_scope = {
			enable = true
		},
		smart_rename = {
			enable = true,
			keymaps = {
				smart_rename = "grr"
			}
		},
		navigation = {
			enable = true,
			keymaps = {
				goto_definition = "gnd",
				list_definitions = "gnD"
			}
		}
	},
	textobjects = {
		enable = true,
		keymaps = {
			["af"] = "@function.outer",
			["if"] = "@function.inner",
			["aC"] = "@class.outer",
			["iC"] = "@class.inner",
			["ac"] = "@conditional.outer",
			["ic"] = "@conditional.inner",
			["ae"] = "@block.outer",
			["ie"] = "@block.inner",
			["al"] = "@loop.outer",
			["il"] = "@loop.inner",
			["is"] = "@statement.inner",
			["as"] = "@statement.outer",
			["ad"] = "@comment.outer",
			["am"] = "@call.outer",
			["im"] = "@call.inner"
		}
	},
	ensure_installed = "all"
}
```

### PR DESCRIPTION
**Pull request by vigoux**
Attempt to fix #272 .

### CODE REVIEW

**Comment by steelsojka** [comment]
I tested this and it doesn't break my setup (which works fine prior to this fix). I think the main difference is in my setup I'm calling `setup` in my init.vim prior to `init` being called by the plugin. I believe the latter is happening in the original issue. So, the calls that are added here are a no-op in my case.

**Comment by vigoux** [comment]
Yes I guess that the problem is this. In any case, this needs a fix, I will be pushing a better version soon.

File: lua/nvim-treesitter/configs.lua [hunk 1]
```diff
@@ -278,6 +278,16 @@ function M.setup(user_data)
...
+
+      recurse_modules(function(mod_name)
+        if data.enable then
+          enable_all(mod_name)
+        end
+
+        for _, lang in ipairs(data.disable or {}) do
+          disable_mod_conf_autocmd(mod_name, lang)
+        end
+      end, config.modules, "")
```
Comment (steelsojka) [inline lua/nvim-treesitter/configs.lua:290]: nitpick, you don't need the empty string argument. It should default to it.

Comment (vigoux) [reply lua/nvim-treesitter/configs.lua:290]: Oh nice, give me a sec.

**Pull request merged**